### DISCUSSION TITLE
Deleted a workspace but Turborepo still trying to read its package.json

### DISCUSSION THREAD

**Discussion by danielhep** [vercel/turborepo#7186] [Help] [answered]
I deleted a workspace from my pnpm Turborepo project, but now whenever I run any `turbo run` command I get the following error:
` ERROR  run failed: unable to parse workspace package.json: unable to read package.json: failed to open file '/Users/danielhep/git/otp-frontend/packages/leg-chip/package.json'`

I searched my entire project for references to the `leg-chip` workspace but nothing comes up. I don't know where it's coming from!

**Comment by anthonyshew** [comment]
🤔 Have you re-installed with your package manager to make sure node_modules is correct?

If this doesn't help, could you open an issue with reproduction steps?

**Comment by kickbk** [comment]
Got the same issue with a renamed package. Downgrading to 1.12.2 helped.

**Comment by dominx99** [comment]
Got the same issue with a renamed package. Version: 1.12.2

**Comment by svitlychnyi** [comment]
Deleting node_modules and reinstalling turbo solved problem for me.

**Comment by danielhep** [comment, marked as answer]
I was finally able to fix this. I found some documentation that I can't find again that mentioned not putting internal monorepo dependencies inside the package.json at the top level of the repo. I checked and sure enough, at some point I had added some internal dependencies there. Removing them and rerunning `pnpm install` fixed the issue for me. I am not sure what the linkage is between this mistake on my part and the issue I was encountering- some kind of caching thing? 

Given how obscure this bug was and difficult to track down, I think there should be some checks in turborepo that warn people who put internal dependencies at the top level. @anthonyshew

**Reply by dewbjorn** [reply]
Interesting. In my case everything was installed internally where it was meant to be, and I still got this error. Only downgrading to 1.12.2 helped.

**Reply by kickbk** [reply]
It is possible that downgrading to 1.12.2 wasn't what enabled turbo to work. It is possible that something in its cache gets reset when a new version is installed.

**Reply by anthonyshew** [reply]
https://github.com/vercel/turbo/issues/7245#issuecomment-1926199995

We're thinking this has more to do with our daemon. We're looking into it.

**Comment by davidfurlong** [comment]
that doesn't fix my case. My project uses yarn as well

**Comment by davidfurlong** [comment]
I solved this by deleting my folder and re cloning

**Reply by leamandeep** [reply]
I tried everything from deleting node modules to downgrading turborepo version. Nothing is working properly for me
